src/react/utils.ts

```diff
@@ -0,0 +1,11 @@
+export function shouldThrowError<TError>(
+  suspense: boolean | undefined,
+  _useErrorBoundary: boolean | ((err: TError) => boolean) | undefined,
+  error: TError
+): boolean {
+  if (typeof _useErrorBoundary === 'function') {
+    return _useErrorBoundary(error)
+  }
+
+  return !!suspense || !!_useErrorBoundary
```

Comment:
@TkDodo  These specs are great and straightforward with no breaking changes!
Thank you @dantman

Reply:
> my suggestion would be that `useErrorBoundary` defaults to `false` if it's `undefined` and we have `suspense` on, but it should still respect if `false` is explicitly passed into it. So, in summary:

I agree that it would be nice for it to work that way. And we may want to think about that long term for a future major release.

However it's possible that a current project has set `useErrorBoundary: false` in their Query Client's defaultOptions and then set `suspense: true` for a single query. In that case this change would result in a breaking change in that query, changing from the error being thrown to the error not being thrown after the user does a minor package upgrade.

Reply:
That’s a good point, the docs also say that it defaults to false afaik 🤔

Reply:
okay, for backwards compatibility, I concur that we shouldn't change it. But I'm still very much on the fence about the different behaviour between `false` and `() => false` when it comes to suspense being enabled. This will be very confusing and hard to explain :/

Reply:
> In that case this change would result in a breaking change in that query, changing from the error being thrown to the error not being thrown.

Right now `useErrorBoundary: false` or `useErrorBoundary: true`  have no effect when `suspense: true`, so we can see this change as a fix. 
Setting `useErrorBoundary: false` in the defaultOptions seems to be something pretty rare because  `useErrorBoundary` is `false` by default (in the docs).

Reply:
> Setting useErrorBoundary: false in the defaultOptions seems to be something pretty rare because useErrorBoundary is false by default (in the docs).

this sounds convincing. a quick search on github shows that there are some projects that set `useErrorBoundary: false` on the global config, but I doubt that they set `suspense: true` at the same time. At least I haven't seen it, and I think respecting the flag is better than ignoring it.

@tannerlinsley what's your take please?